src/test/javascript/spec/component/overview/exercise-details/course-exercise-details.component.spec.ts

```diff
@@ -211,24 +210,24 @@ describe('CourseExerciseDetailsComponent', () => {
         studentParticipation.exercise = exercise;
 
         const exerciseDetail = { ...exercise, studentParticipations: [studentParticipation] };
-        const exerciseDetailReponse = of({ body: exerciseDetail });
+        const exerciseDetailResponse = of({ body: exerciseDetail });
 
         // return initial participation for websocketService
-        stub(participationWebsocketService, 'getParticipationForExercise').returns(studentParticipation);
-        stub(complaintService, 'findBySubmissionId').returns(of({} as EntityResponseType));
+        jest.spyOn(participationWebsocketService, 'getParticipationForExercise').mockReturnValue(studentParticipation);
+        jest.spyOn(complaintService, 'findBySubmissionId').mockReturnValue(of({} as EntityResponseType));
 
-        mergeStudentParticipationStub.returns(studentParticipation);
+        mergeStudentParticipationMock.mockReturnValue(studentParticipation);
         const changedParticipation = cloneDeep(studentParticipation);
         const changedResult = { ...result, id: 2 };
         changedParticipation.results = [changedResult];
-        subscribeForParticipationChangesStub.returns(new BehaviorSubject<Participation | undefined>(changedParticipation));
+        subscribeForParticipationChangesMock.mockReturnValue(new BehaviorSubject<Participation | undefined>(changedParticipation));
 
         fixture.detectChanges();
         tick(500);
         expect(comp).to.be.ok;
 
-        // override stub to return exercise with participation
-        getExerciseDetailsStub.returns(exerciseDetailReponse);
+        // override mock to return exercise with participation
+        getExerciseDetailsMock.mockReturnValue(exerciseDetailResponse);
```

Comment:
It says here override. I think it doesn't make sense then to define it globally.

Reply:
I must say I am not sure how all of these definitions are working together. I'd prefer to not try to refactor minor parts without fully understanding the tests, but refactoring the whole file seems out of scope